packages/core/src/utilities/frameRangeUtils.ts

```diff
@@ -0,0 +1,56 @@
+import { triggerEvent, eventTarget } from '@cornerstonejs/core';
+
+export type FramesRange = [number, number] | number;
+
+/**
+ * This class handles the annotation frame range values for multiframes.
+ * Mostly used for the Video viewport, it allows references to
+ * a range of frame values.
+ */
+export default class FrameRangeUtils {
+  protected static frameRangeExtractor =
+    /(\/frames\/|[&?]frameNumber=)([^/&?]*)/i;
+
+  protected static imageIdToFrames(imageId: string): FramesRange {
+    const match = imageId.match(this.frameRangeExtractor);
+    if (!match || !match[2]) {
+      return null;
+    }
+    const range = match[2].split('-').map((it) => Number(it));
+    if (range.length === 1) {
+      return range[0];
+    }
+    return range as FramesRange;
+  }
+
+  public static multiframeImageId(imageId: string, frameNumber = 1) {
```

Comment:
multiframeImaegIdToFrames? do you need this extra? can't you make the `imageIdToFrames` handle multiframe case too?